src/screens/Asset/ReceiveModal.js

```diff
@@ -138,79 +81,53 @@ const ReceiveModal = ({
-  const needsSmallButtons = showBuyTokensButton && SCREEN_WIDTH < 300;
   const profileImageURI = profileImage ? `${profileImage}?t=${lastUpdateTime}` : null;
 
   return (
     <SlideModal
       onModalHide={onModalHide}
       noPadding
       noClose
-      headerLeftItems={showErc20Note ? [{
-        custom: (
-          <LabelBadge
-            label={t('label.erc20TokensOnly')}
-            labelStyle={{ fontSize: fontSizes.tiny }}
-            primary
-          />
-        ),
-      }] : undefined}
+      // TODO : The label itself will be revisited later on the direction of the product team
+      // headerLeftItems={showErc20Note ? [{
+      //   custom: (
+      //     <LabelBadge
+      //       label={t('label.erc20TokensOnly')}
+      //       labelStyle={{ fontSize: fontSizes.tiny }}
+      //       primary
+      //     />
+      //   ),
+      // }] : undefined}
       centerFloatingItem={
         <ImageWrapper style={{ position: 'absolute', marginTop: -24 }}>
-          <ProfileImage
-            uri={profileImageURI}
-            userName={username}
-            diameter={48}
-          />
+          <ProfileImage uri={profileImageURI} userName={username} diameter={48} />
         </ImageWrapper>
       }
     >
       <ContentWrapper forceInset={{ top: 'never', bottom: 'always' }}>
-        <WarningBanner rounded small />
-        {!!ensName && (
-          <InfoView>
-            <BaseText
-              big
-              onPress={() => handleCopyToClipboard(ensName, true)}
-              center
-            >
+        <InfoView>
+          {!!ensName && (
+            <TextWithCopy textToCopy={ensName} iconColor="#007aff">
               {ensName}
-            </BaseText>
-            <BaseText regular center secondary>{t('label.yourEnsName')}</BaseText>
-          </InfoView>
-        )}
-        <QRCodeWrapper>
-          <View style={{ overflow: 'hidden', padding: 10 }}>
-            {!!address && <QRCodeWithTheme value={address} size={160} />}
-          </View>
-          <WalletAddress onPress={() => handleCopyToClipboard(address)}>
-            {address}
-          </WalletAddress>
-        </QRCodeWrapper>
-        <ButtonsRow>
-          {showBuyTokensButton && (
-          <Button
-            title={t('button.buyTokens')}
-            onPress={handleBuyTokens}
-            primarySecond
-            small={needsSmallButtons}
-            style={{ flex: 1, marginRight: 10 }}
-          />
+            </TextWithCopy>
           )}
-          <Button
-            title={t('button.shareAddress')}
-            onPress={handleAddressShare}
-            small={needsSmallButtons}
-            style={{ flex: 1 }}
-          />
-        </ButtonsRow>
-        {showBuyTokensButton && (
-        <IconsContainer>
-          <Image source={visaIcon} />
-          <IconsSpacing />
-          <Image source={mastercardIcon} />
-        </IconsContainer>
-        )}
+          <WalletAddress>{address}</WalletAddress>
```

Comment:
Does the <WalletAddress> component also copy to clipboard when tapped?

Reply:
@IAmKio No, since in the design there was copy address button, so WalletAddress is not designed in such a way. Do we need to change it and make it copy to clipboard when tapped?